### DISCUSSION TITLE
Meetup & Discussion

### DISCUSSION THREAD

**Discussion by ZuitAMB** [agrlh/fininsights#11] [General]
A small meetup for exchanging visions etc. could help? E.g. on Discord. We should also discuss project structure, git structure, how we would discuss ideas, push code.

**Comment by ZuitAMB** [comment]
Discussions and Issues are more convenient for exchange, compared to the Trello-like project/ToDo section

We can do lists here

- A
- B
- C

Or track progress:

- [ ] A
- [ ] B
- [ ] C

And reference Issues, Branches and so on^^

**Comment by KhanPhelan** [comment]
![image](https://user-images.githubusercontent.com/87005254/125384365-1a8ece00-e399-11eb-8e23-0b98b74b7a39.png)
So far i used this as a template (i've added some more fields in the current version but you get a picture)

I would say the easiest part is Finviz?

**Reply by ZuitAMB** [reply]
I would recommend to separate the project into multiple parts, for example like this:
![FinInsights](https://user-images.githubusercontent.com/56884080/125493372-578b494e-79d9-4480-b16d-df0b61a4f659.png)
We could use Dockerized programs for the different parts. Most of the your proposed fundamentals should be available from multiple APIs/Sources. For example, Yahoo Finance should also offer most of them. We should consider using multiple (redundant) data sources, as webscraping the public data could lead to them blacklisting our used IP after too many requests. Some APIs offer e.g. 300 free requests per day, and so on. Additionally, there might be some legal restraints depending on which the API would have to be exchanged.

> Estimated Intrinsic Value

Do you mean in the sense of book value? Value of a discounted cash flow analysis? Or some other custom valuation (including e.g. inflation, peer companies, growth expectations, ESG score, and so on)?

The finviz API should be easily accessible e.g. with the [FinvizFinace python package](https://pypi.org/project/finvizfinance/), but Yahoo finance offers similar support with the [yFinance package](https://pypi.org/project/yfinance/). To my knowledge the official  Seeking Alpha API requires a premium subscription ~30€/month - but scraping might also be possible. The majority of features could be calculated based on (historic) stock prices and a recent balance sheet.
But I guess storing and using the data also shouldn't be that difficult 😄 And a [discord bot](https://realpython.com/how-to-make-a-discord-bot-python/) could be fun.

**Reply by KhanPhelan** [reply]
Looks very good!
Yeah for now i use this manual IV Calculation:
https://docs.google.com/spreadsheets/d/1OD-vYzsqfCLfsr0qzMojlMw0mXiaezxCsMrxfMQ5c0w/edit#gid=713559401
With this for the Sector PE-Ratio:
http://pages.stern.nyu.edu/~adamodar/New_Home_Page/datafile/pedata.html
And sometimes this:
https://fullratio.com/stocks

Again - could be easily included. So i would say the first step would be taking the Yahoo or Finviz API and retrieve as many attributes as we can, and from there we can expand to more complex things like 
Cash-To-Debt 
Equity-to-Asset 
Debt-to-EBITDA 
Piotroski F-Score
Altman Z-Score
Beneish M-Score 

etc

**Comment by dsaoijhlgfdngfsd** [comment]
Hi guys, going through the ongoing discussions. From what I see I understand you plan to pull data from external sources and do some analysis / maybe prediction on them. Since I wasn't here from the start, I don't have the whole picture -  maybe some questions:
1) is this planned to be a "service" (from cloud) or something user spins up on his computer?
2) what's the end goal? I see a trading bot in the diagram above. Do you plan to actually support algo trading?
3) Re frontend, are we talking a full scale web application?

**Reply by ZuitAMB** [reply]
I am not sure about the full picture - I personally am interested in algorithmic trading/automated investing, the Reddit OP mentioned some webservice/discord bot - so therefore I suggested these three use cases in the diagram. If we have some stock analysis functionality, we can use the data for all of the more advanced use cases.

The first step would be a local program/script, but once we e.g. have a docker container with some API or front-end, deploying it on a webserver/cloud should be an option. :)

To grasp the general scope, I would prefer an (online) meetup (or discussion in here).

**Reply by KhanPhelan** [reply]
Morning @dsaoijhlgfdngfsd and @ZuitAMB 
Yeah sorry about the inactiveness - the online-exams i have to supervise with my group will start in the next days/weeks and  am bit busy - once the started / during august - i will have more time for that.
A meetup would be very good - due to our different time-zones the only possible way would be after market close 4pm EST / 22.00 GMT+1.

So as a first step:
 a (online/cloud based) script/service to pull and display fundamentals/financials from a given stock with some higher calculation from thereon

i would love to build this into a discord bot.
From there, we can of course wander in the fields of algo trading - although will need a bit of brain power to identify financial metrics that are reliable and worth trading of

**Reply by dsaoijhlgfdngfsd** [reply]
Sure, let's do a (virtual) meetup and align on the goals. Will be off for the rest of the week, but starting Monday I should be available always after market close.

**Reply by ZuitAMB** [reply]
So August 2nd, 22 GMT+1 ?

**Reply by dsaoijhlgfdngfsd** [reply]
Hi @KhanPhelan - what's your availability? Still busy with the exams? When would the meetup suit you? Also, what platform should we use to meet up?

**Reply by ZuitAMB** [reply]
Discord is my platform of choice - but I am fine with any other platform. Zoom, jitsi, Teams, Hangouts

**Reply by KhanPhelan** [reply]
> Hi @KhanPhelan - what's your availability? Still busy with the exams? When would the meetup suit you? Also, what platform should we use to meet up?

Hi ry, Stress. Uhm Thursday evening would be possible?
Discord, Zoom, Teams, whatever you want

**Reply by ZuitAMB** [reply]
> Thursday evening

👍

**Reply by dsaoijhlgfdngfsd** [reply]
Should I set up some discord server, or are you planning to meet on any existing server?

**Reply by ZuitAMB** [reply]
A new server should be alright.

**Reply by fynnfluegge** [reply]
Hey guys, thanks for the invitation🙌 had zero time the last weeks for dive into here, from now on i am available

**Reply by ZuitAMB** [reply]
Hey Fynn, our first step would be checking out the different available APIs and data sources to extract fundamental, technical data, metrics etc. as outlined in https://github.com/agrlh/fininsights/discussions/11#discussioncomment-996614 by @KhanPhelan, who might post an updated list soon.
Our first goal would be a script for collecting and presenting the data for around 25 tickers, but expand it later. :)

**Reply by fynnfluegge** [reply]
alright! im not that familiar with python yet. I'm experienced in Kotlin/Java and Angular and cloud stuff. Will dive into python soon. If we need backend backend services with spring boot i am your man. I have already a decent angular frontend for monitoring several tickers. It was running in a kube cluster with Google Cloud but my free trial was used up some days ago. I will deploy it at another cloud provider soon and share the link here :)

**Reply by fynnfluegge** [reply]
hey guys!
I deployed my application https://github.com/fynnfluegge/stonk-platform-ng-kt to the cloud here
http://47.91.78.219:3000

It is running at Alibaba cloud. Why baba cloud? Because I got 12 month free linux machine, but with only 1 mbit download. So the first access could be a bit slow, even it is compressed with gzip. I used google cloud in the past, but my 300$ free amount melted, since a ran a kubernetes cluster there. Alibaba is fine for just playing around with Frankfurt location but I plan to use azure soon if it provides more free download bandwith.

**Reply by fynnfluegge** [reply]
Hello guys, does anyone have experience with sharadar? https://www.quantrocket.com/pricing/data/sharadar/
They provide US company fundamentals from 1990 till present.
